### PR TITLE
Newbie commit

### PR DESCRIPTION
**Pull request by soupman99**
I'm pretty new at github but did some work on the generator templates.  If there's something I need to change please let me know.  Hopefully part of this is usable.

All components needed to create a full CRUD module using `yo meanjs:crud-module <moduleName>`

### CODE REVIEW

**Comment by mleanos** [comment]
@soupman99 Thank you for helping out here. 

To make it easier to review, can you provide a summary of these changes? Have you tested this?

@codygaig How is this related to the "Remove Templates" PR?

**Comment by soupman99** [comment]
@mleanos it's pretty much a more complete build out of #131 using the articles module as a template.  Essentially it's all the components you'd need to create a new full module.  I haven't written any karma tests, but I have been testing as I went along.  Everything seems to be working as expected.

**Comment by mleanos** [comment]
@soupman99 Can you provide screen shots? I'll try to pull this down locally and test, over the weekend.

File: crud-module/index.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                    pluralModuleName: this.pluralModuleName,
+                    pluralCapitalizedModuleName: this.pluralCapitalizedModuleName
+                }
+            );
+        }
+    },
+
+    removeAngularModule: function () {
+        if(!this.addAngularModule) {
+            exec('rm -rf ' + folderPath + 'client')
```
Comment (mleanos) [inline crud-module/index.js]: `rm -rf` won't work on Windows, as I recall. I'm actually on windows but I have *cygwin* installed, so it works for me. I can't remember what the resolution to this was previously with the main generator.

@codydaig How was this reconciled in the current generator?

Comment (ilanbiala) [inline crud-module/index.js]: Use `fs` or `this.fs` if possible.

**Comment by ilanbiala** [comment]
Don't change permissions please for files. Also, we require tests to merge, and that will also allow us to test it constantly against other features we will add, so please add tests.

**Comment by codydaig** [comment]
@soupman99 Thanks for contributing! Few things that need to be cleaned up: don't use the child_process.exec command to manage files. Use the internal `fs` module, we also need tests to ensure that it was generated as expected.

**Comment by soupman99** [comment]
@mleanos I'm not sure I understand what you mean by screenshots? You mean of the html templates?

@ilanbiala I didn't realize I had changed file permissions.  I'm new to github. Could you be a little more specific?

**Comment by mleanos** [comment]
@soupman99 I meant screen shots of you using the generator to create a new module. It's one way for us to see it in action, other than pulling this branch down locally. I plan to test myself, but it would be nice for others to be able to see it.

**Comment by ilanbiala** [comment]
@soupman99 we've merged in the remove templates PR, and we'd like to include this or #131 in the next release very soon. Can you double check to see if anything in this PR needs to be changed. @codydaig functionally does it LGTY?

File: crud-module/index.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                .catch(function (err) {
+                    log.red(err);
+                    return;
+                });
+        }
+    },
+
+    removeExpressModule: function () {
+        if(!this.addExpressModule) {
+            exec('rm -rf ' + folderPath + 'server')
```
Comment (codydaig) [inline crud-module/index.js]: `rm` won't work on Windows

File: package.json [hunk 3]
```diff
@@ -31,7 +31,10 @@
   "dependencies": {
     "bluebird": "~2.10.0",
     "chalk": "~1.1.1",
+    "mkdirp": "~0.5.1",
+    "underscore": "~1.8.3",
     "underscore.string": "~3.2.2",
+    "underscore.inflection": "~1.3.0",
```
Comment (codydaig) [inline package.json]: Remove

File: crud-module/index.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
+var Promise = require('bluebird'),
+    child_process = require('child_process'),
+    s = require('underscore.string'),
+    inflection = require( 'underscore.inflection' ),
```
Comment (codydaig) [inline crud-module/index.js]: This isn't used anywhere. Please remove.

**Comment by codydaig** [comment]
@ilanbiala I left some more line comments that need too get addressed. The commits also need too be squashed. I haven't tested the code yet, but if it works, then we need too write at least one test for these. (I'm willing too do so after this gets merged before the release.) 

@soupman99 When do you think you'll be able too make these changes? We're hoping too see this in the release coming this week.

**Comment by soupman99** [comment]
I don't know if I'll be able to make the changes by then.



-Justin
youn2493@gmail.com
718.679.3231

File: crud-module/index.js (hunk 1 above)
Comment (SharkIng) [inline crud-module/index.js]: rm -rf is a dangerous command so I dont think its a good idea to use it here

File: crud-module/index.js (hunk 2 above)
Comment (ilanbiala) [inline crud-module/index.js]: @codydaig maybe we should just write another lib file for working with fs that does stuff like remove file/folder cleanly?

File: crud-module/index.js (hunk 1 above)
Comment (aneeshd16) [inline crud-module/index.js]: https://github.com/isaacs/rimraf is a platform independent alternative for rm -rf

**Comment by ilanbiala** [comment]
@codydaig is this PR still relevant or are we going to have this functionality with @UndeadBaneGitHub's PRs?

**Comment by UndeadBaneGitHub** [comment]
@ilanbiala I haven't PRed full CRUD generator, so, I suppose, this is relevant. 
However, tomorrow I plan on fixing the remaining issues with my remaining PRs and can PR CRUD module, built like my other generators for consistency.

**Comment by ilanbiala** [comment]
@UndeadBaneGitHub we still need a CRUD generator, right?

**Comment by UndeadBaneGitHub** [comment]
Yep. I have a strong feeling I will write it sooner, than existing PR would
get fixed

**Comment by ilanbiala** [comment]
@UndeadBaneGitHub that would be great.

**Comment by ilanbiala** [comment]
@UndeadBaneGitHub this can be closed, right?

**Comment by UndeadBaneGitHub** [comment]
@ilanbiala yup, it can be.

**Pull request closed without merging**